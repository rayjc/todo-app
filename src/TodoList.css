.TodoList-title {
    letter-spacing: 0.2rem;
}

.TodoList-header {
    background: azure;
    padding: 50px;
    border-bottom: 2px solid lightgray;
}