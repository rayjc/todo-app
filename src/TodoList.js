import React, { useState } from 'react';
import { v4 as uuidv4 } from 'uuid';
import NewTodoForm from './NewTodoForm';
import Todo from './Todo';
import './TodoList.css';


const TodoList = () => {
  const [todos, setTodos] = useState([]);

  const addTodo = (name) => {
    setTodos([...todos, { name, isDone: false, id: uuidv4() }]);
  }

  const removeTodo = (id) => {
    setTodos(todos.filter(todo => todo.id !== id));
  }

  return (
    <div className="TodoList">
      <div className="TodoList-header">
        <h1 className="TodoList-title">Todo Tracker</h1>
        <NewTodoForm addTodo={addTodo} />
      </div>
      <div>
        {
          todos.map(({ id, name, isDone }, idx) =>
            <Todo key={id} id={id} name={name} isDone={isDone} removeTodo={removeTodo} />
          )
        }
      </div>
    </div>
  )
}


export default TodoList;