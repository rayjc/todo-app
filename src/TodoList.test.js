import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

it("renders without crashing", function() {
  render(<TodoList />);
});

it("should match default snapshot", function() {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
})

it("can add a new todo", function() {
  const { getByLabelText, getByText, queryByText } = render(<TodoList />);

  // no todos yet
  expect(queryByText("test todo")).not.toBeInTheDocument();

  const todoInput = getByLabelText("Task:");
  const submitBtn = getByText("Add");

  // fill out the form
  fireEvent.change(todoInput, { target: { value: "test todo" } });
  fireEvent.click(submitBtn);

  // todo exists!
  expect(queryByText("test todo")).toBeInTheDocument();
});

it("can cross out a todo", function() {
  const { getByRole, getByLabelText, getByText, queryByText } = render(<TodoList />);
  const todoInput = getByLabelText("Task:");
  const submitBtn = getByText("Add");

  // fill out the form
  fireEvent.change(todoInput, { target: { value: "test todo" } });
  fireEvent.click(submitBtn);

  const checkBox = getByRole("checkbox");

  // crossout todo
  fireEvent.click(checkBox);
  expect(queryByText("test todo")).toHaveClass("completed")
});

it("can remove a todo", function() {
  const { getByRole, getByLabelText, getByText, queryByText } = render(<TodoList />);
  const todoInput = getByLabelText("Task:");
  const submitBtn = getByText("Add");

  // fill out the form
  fireEvent.change(todoInput, { target: { value: "test todo" } });
  fireEvent.click(submitBtn);

  const checkBox = getByRole("checkbox");

  // crossout todo to show remove button
  fireEvent.click(checkBox);

  const removeBtn = getByText("x");
  fireEvent.click(removeBtn);

  expect(queryByText("test todo")).not.toBeInTheDocument();
});
