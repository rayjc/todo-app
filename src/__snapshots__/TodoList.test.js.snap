// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match default snapshot 1`] = `
<DocumentFragment>
  <div
    class="TodoList"
  >
    <div
      class="TodoList-header"
    >
      <h1
        class="TodoList-title"
      >
        Todo Tracker
      </h1>
      <form>
        <label
          class="NewTodoForm-label"
          for="todo-input"
        >
          Task:
        </label>
        <input
          id="todo-input"
          name="todo"
          type="text"
          value=""
        />
        <button
          type="submit"
        >
          Add
        </button>
      </form>
    </div>
    <div />
  </div>
</DocumentFragment>
`;
